mobile/lib/utils/version_compatibility.dart

```diff
@@ -0,0 +1,17 @@
+(bool, String, int) checkVersionCompatibility(
+  int appMajor,
+  int appMinor,
+  int serverMajor,
+  int serverMinor,
+) {
+  if (serverMajor != appMajor) {
+    return (false, 'major', appMajor);
+  }
+
+  // Add latest compat info up top
+  if (serverMinor < 106 && appMinor >= 106) {
+    return (false, 'minor', 106);
+  }
+
+  return (true, '', 0);
```

Comment:
This return tuple is quite ugly imo. Maybe this should be called something like `getVersionCompatabilityMessage()` and returns a string or null instead.